### PR TITLE
New post: "JavaScript The Language:  Async/Await"

### PR DESCRIPTION
**Pull request by HeeL**

### CODE REVIEW

**Comment by ViktorZener** [comment]
great :+1: 
just have a small suggestion, when you run it('..... -> start with should, then a proper sentence while running tests is generated. describe -> it :=> which produces for instance `async/await should deliver a promise`

File: _posts/2020-02-19-js-the-language-meetup.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+date: 2020-02-19 10:00:00 +0200
+categories: javascript
+author_names: ["Pablo Palacios", "Sergii Paryzhskyi"]
+read_time : 5
+show_related_posts: false
+square_related:
+feature_image: posts/2020-02-19-js-the-language-meetup-async-await/poster.jpg
+---
+
+Yesterday we've hosted a “JavaScript the Language” Meetup at HolidayCheck. This is a monthly meetup where we go into details about specific aspects of JavaScript. The topic of Thursday's meetup was Async/Await.
```
Comment (singles) [inline _posts/2020-02-19-js-the-language-meetup.md]: Later you're using the same wording "Async/await", but with lower case "async/await". I would suggest sticking to one form. Or maybe even `async`/`await` (or simpler `async/await`) using code syntax to emphasize that it's a language construct?

File: _posts/2020-02-19-js-the-language-meetup.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+There is no presentations, but it is a very interactive and hands-on event. Instead of a keynote speaker and slides presentation, the
+guests here are protagonists. As usually it all starts with a socialising, and get to know each other while enjoying snacks and drinks:
+
+<img src="/img/posts/2020-02-19-js-the-language-meetup-async-await/socializing.jpg" alt="Socializing" class="centered" />
+
+After having some food and drinks, we start an introduction round. Everything is very informal and its a good start to make people talk. Normally people say where they came from or share experience they've made with JavaScript so far. That was the point as we've realized how broad our audience actually is. Some people had barely started with programming, some others were actively using JavaScript over 15 years. That turned out not to be a problem at all. In order to bring everyone on the same page about async/await, we've started by reminding ourself the general knowledge about this language construct and what it is normally used for. Afterward we've started to collect all the things that people would be interested to know in context of async/await:
```
Comment (pablopalacios) [inline _posts/2020-02-19-js-the-language-meetup.md]: ourself -> ourselves

Comment (singles) [inline _posts/2020-02-19-js-the-language-meetup.md]: ``s/its/it`s``

File: _posts/2020-02-19-js-the-language-meetup.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+
+<img src="/img/posts/2020-02-19-js-the-language-meetup-async-await/ideas-list.jpg" alt="Ideas List" class="centered" />
+
+We've tried to answer, present and learn through a dynamic code session. We've started by checking what can and cannot be "awaited". Should we use just async functions as await expressions? Starting from that, we've tried to await almost everything (sync functions, values, generators, code blocks, exceptions and so on). That was lots of fun and many wow-effects, especially when a code-execution went not as many people would expect it to go.
+
+Then, we focused on the relationship between async functions and promises. We've covered topics such as implicit promises, error handling and chaining of async functions.
+
+As the last test, we've checked the behavior of generators as an await expression. Nobody was certain if it returns a yielded value from the generator or will it return the generator itself.
+
+The test cases for all arisen questions can be found in the code snippet below this blogpost. There you can take a deep look on all the experiments we did.
```
Comment (singles) [inline _posts/2020-02-19-js-the-language-meetup.md]: I think it should be `s/blogpost/blog post`, not sure though 🤔

File: _posts/2020-02-19-js-the-language-meetup.md [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+
+<img src="/img/posts/2020-02-19-js-the-language-meetup-async-await/next-topics.JPG" alt="Next Topics" class="centered" />
+
+This wishlist was filled within a couple of minutes as people started to name all topics that they would like to learn or know more about. Here we've experienced again a high participation rate of all people no matter how long they have been using JavaScript. The final list of all topics looked liked this:
+
+<img src="/img/posts/2020-02-19-js-the-language-meetup-async-await/next-topics-list.jpg" alt="Next Topics List" class="centered" />
+
+Everyone had two votes that he or she could give for any topic from the list. At first it looks very promising for `Prototypes`, but then the clear winner was `Reflect`, which got the most of the votes. We are excited about hosting this event in March and looking forward to see you all at our office!
+
+```javascript
```
Comment (singles) [inline _posts/2020-02-19-js-the-language-meetup.md]: How about putting this code snippet higher? I mean, code discusses `async/await` and mentions, that there's a code snippet at the end. Then you start to write about future meetup, we read about Prototypes and `Reflect`, and 1 sentence later I see code example for `async/await`. That was unexpected and IMO unnecessary context switch.

File: _posts/2020-02-19-js-the-language-meetup.md [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+author_names: ["Pablo Palacios", "Sergii Paryzhskyi"]
+read_time : 5
+show_related_posts: false
+square_related:
+feature_image: posts/2020-02-19-js-the-language-meetup-async-await/poster.jpg
+---
+
+Yesterday we've hosted a “JavaScript the Language” Meetup at HolidayCheck. This is a monthly meetup where we go into details about specific aspects of JavaScript. The topic of Thursday's meetup was Async/Await.
+
+There is no presentations, but it is a very interactive and hands-on event. Instead of a keynote speaker and slides presentation, the
```
Comment (mkrmpotic) [inline _posts/2020-02-19-js-the-language-meetup.md]: Should be 'There are no presentations'

File: _posts/2020-02-19-js-the-language-meetup.md [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+guests here are protagonists. As usually it all starts with a socialising, and get to know each other while enjoying snacks and drinks:
```
Comment (mkrmpotic) [inline _posts/2020-02-19-js-the-language-meetup.md]: I think it should be 'it all starts with socialising, and getting to know each other'

File: _posts/2020-02-19-js-the-language-meetup.md (hunk 2 above)
Comment (mkrmpotic) [inline _posts/2020-02-19-js-the-language-meetup.md]: Should be 'and it is a good start', or 'it's'.

I would rewrite 'That was the point as we've realized how broad our audience actually is.' to 'At that point we have realized how broad our audience actually was.'

Something like this maybe? Not sure... ', some others were actively using JavaScript over 15 years.' -> ', while there were some that are actively using JavaScript for over 15 years.'

File: _posts/2020-02-19-js-the-language-meetup.md [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+
+After having some food and drinks, we start an introduction round. Everything is very informal and its a good start to make people talk. Normally people say where they came from or share experience they've made with JavaScript so far. That was the point as we've realized how broad our audience actually is. Some people had barely started with programming, some others were actively using JavaScript over 15 years. That turned out not to be a problem at all. In order to bring everyone on the same page about async/await, we've started by reminding ourself the general knowledge about this language construct and what it is normally used for. Afterward we've started to collect all the things that people would be interested to know in context of async/await:
+
+<img src="/img/posts/2020-02-19-js-the-language-meetup-async-await/collect-ideas.jpg" alt="Collect ideas" class="centered" />
+
+Process of collecting ideas and questions went quite actively and proposals were coming from participants of all levels. It seems like it took us less than 10 minutes to fill up the flipchart with lots of exciting topics to tackle:
```
Comment (mkrmpotic) [inline _posts/2020-02-19-js-the-language-meetup.md]: Maybe something like 'Process of collecting ideas and questions went quite actively and proposals were coming from participants of all levels.' -> 'Participants of all experience levels were actively engaged in proposing ideas and questions to be discussed in more detail.'

File: _posts/2020-02-19-js-the-language-meetup.md [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+
+<img src="/img/posts/2020-02-19-js-the-language-meetup-async-await/ideas-list.jpg" alt="Ideas List" class="centered" />
+
+We've tried to answer, present and learn through a dynamic code session. We've started by checking what can and cannot be "awaited". Should we use just async functions as await expressions? Starting from that, we've tried to await almost everything (sync functions, values, generators, code blocks, exceptions and so on). That was lots of fun and many wow-effects, especially when a code-execution went not as many people would expect it to go.
```
Comment (mkrmpotic) [inline _posts/2020-02-19-js-the-language-meetup.md]: Should we use just async -> Should we simply use async...

File: _posts/2020-02-19-js-the-language-meetup.md [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+
+Then, we focused on the relationship between async functions and promises. We've covered topics such as implicit promises, error handling and chaining of async functions.
+
+As the last test, we've checked the behavior of generators as an await expression. Nobody was certain if it returns a yielded value from the generator or will it return the generator itself.
+
+The test cases for all arisen questions can be found in the code snippet below this blogpost. There you can take a deep look on all the experiments we did.
+
+Finally, it is important to say that the topic for the next meetup is
+already chosen. In order to do so, we've started by collecting all ideas that people would be interested to hear about:
```
Comment (mkrmpotic) [inline _posts/2020-02-19-js-the-language-meetup.md]: In order to do so, we have collected all ideas....

File: _posts/2020-02-19-js-the-language-meetup.md [hunk 10]
```diff
@@ -0,0 +1,135 @@
...
+
+<img src="/img/posts/2020-02-19-js-the-language-meetup-async-await/next-topics.JPG" alt="Next Topics" class="centered" />
+
+This wishlist was filled within a couple of minutes as people started to name all topics that they would like to learn or know more about. Here we've experienced again a high participation rate of all people no matter how long they have been using JavaScript. The final list of all topics looked liked this:
```
Comment (mkrmpotic) [inline _posts/2020-02-19-js-the-language-meetup.md]: to name all the topics they would like to...

The final list of the topics looked like...

File: _posts/2020-02-19-js-the-language-meetup.md [hunk 11]
```diff
@@ -0,0 +1,135 @@
...
+
+<img src="/img/posts/2020-02-19-js-the-language-meetup-async-await/next-topics-list.jpg" alt="Next Topics List" class="centered" />
+
+Everyone had two votes that he or she could give for any topic from the list. At first it looks very promising for `Prototypes`, but then the clear winner was `Reflect`, which got the most of the votes. We are excited about hosting this event in March and looking forward to see you all at our office!
```
Comment (mkrmpotic) [inline _posts/2020-02-19-js-the-language-meetup.md]: Each participant had two votes to give for any of the listed topics. At first it looked very promising for 'Protoypes', but in the end the clear winner was....

We are already excited to host this event again in March, and looking forward to seeing all of you at our office!

File: _posts/2020-02-19-js-the-language-meetup.md (hunk 4 above)
Comment (singles) [reply _posts/2020-02-19-js-the-language-meetup.md]: Love the new format, when there's couple of code blocks 👍

**Comment by HeeL** [comment]
Thank you for all your comments and @pablopalacios for following suggestions and improving stuff.
I've updated the publish date and we are ready to ship

**Pull request merged**